### PR TITLE
Workaround to allow http 1.0 responses to successfully be decoded

### PR DESCRIPTION
**Pull request by bretambrose**
Allows for the parsing of http/1.0 responses.  An issue exists to do further investigation on what, if any, additional http1.0 support we need to do.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/test_decode.c [hunk 1]
```diff
@@ -196,11 +196,33 @@ static int s_http_test_request_bad_version(struct aws_allocator *allocator, void
     return AWS_OP_SUCCESS;
 }
 
+AWS_TEST_CASE(http_test_response_old_version, s_http_test_response_old_version);
+static int s_http_test_response_old_version(struct aws_allocator *allocator, void *ctx) {
```
Comment (graebm) [inline tests/test_decode.c]: trivial: http_test_response_1_0

**Comment by graebm** [comment]
> Is this really all there is? The decoding process itself doesn't change at all?

We opened a ticket to investigate the other differences. This is just to unblock us

File: tests/test_decode.c [hunk 2]
```diff
@@ -196,11 +196,32 @@ static int s_http_test_request_bad_version(struct aws_allocator *allocator, void
...
+    aws_http_decoder_destroy(decoder);
+    s_test_clean_up();
+    return AWS_OP_SUCCESS;
+}
+
+AWS_TEST_CASE(http_test_response_unsupported_version, s_http_test_response_unsupported_version);
+static int s_http_test_response_unsupported_version(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+    s_test_init(allocator);
+    const char *msg = "HTTP/1.2 200 OK\r\n\r\n"; /* Note version is implicitly 0.9, no headers, no status, only body*/
```
Comment (ColdenCullen) [inline tests/test_decode.c]: comment says 0.9, code says 1.2

**Comment by ColdenCullen** [comment]
`tests/test_decode.c failed clang-format check. `

**Pull request merged**